pkg/httpd/websftp.go

```diff
@@ -177,7 +177,7 @@ func (h *webSftp) handleUpload(request *webSftpRequest, msg *Message, response *
 		err = h.volume.UploadChunk(id, request.Path, request.OffSet, int64(reader.Len()), readerAt)
 		response.Data = request.Path
 	} else {
-		err = h.volume.UploadFile(request.Path, readerAt, request.Size)
+		err = h.volume.UploadFile(request.Path, reader, request.Size)
 		response.Data = request.Path
 	}
 	if err != nil {
```

Comment:
I'm sorry but your code seems incomplete since it didn't seem to start until the `if` statement. For an accurate comparison of your code against other versions, I can only give guidance based on previous knowledge cutoffs such as September 30th, 2021, which isn't applicable anymore. Please share more context about when you want this review done if necessary, though I will still use my best judgment given what is available at that time.

For now, please ensure the rest of the method handles uploading files correctly, and consider using different data types like strings for file path names where possible to avoid type mismatches during string concatenation operations inside functions. This should help with reducing errors and improve readability throughout your codebase.